internal/gatewayapi/clienttrafficpolicy.go

```diff
@@ -387,10 +387,11 @@ func translateClientIPDetection(clientIPDetection *egv1a1.ClientIPDetectionSetti
 	httpIR.ClientIPDetection = (*ir.ClientIPDetectionSettings)(clientIPDetection)
 }
 
-func translateListenerSuppressEnvoyHeaders(suppressEnvoyHeaders *bool, httpIR *ir.HTTPListener) {
-	if suppressEnvoyHeaders != nil {
-		httpIR.SuppressEnvoyHeaders = *suppressEnvoyHeaders
+func translateListenerHeaderSettings(headerSettings *egv1a1.HeaderSettings, httpIR *ir.HTTPListener) {
+	if headerSettings == nil {
```

Comment:
this logic looks incorrect, `SuppressEnvoyHeaders` is enabled by default unless `headerSettings != nil && headerSettings.EnableEnvoyDebugHeader == true`
this is the one you plan on handling in a follow up PR ?

Reply:
Yes, exactly.

The implementation PR will handle the logic change, this PR is just to get the API in place while still being able to build the project.

Reply:
Sg